src/bootstrap/tool.rs

```diff
@@ -686,21 +688,21 @@ macro_rules! tool_extended {
 // Note: tools need to be also added to `Builder::get_step_descriptions` in `builder.rs`
 // to make `./x.py build <tool>` work.
 tool_extended!((self, builder),
-    Cargofmt, rustfmt, "src/tools/rustfmt", "cargo-fmt", stable=true, {};
+    Cargofmt, rustfmt, "src/tools/rustfmt", "cargo-fmt", stable=true, submodule="rustfmt", {};
     CargoClippy, clippy, "src/tools/clippy", "cargo-clippy", stable=true, in_tree=true, {};
     Clippy, clippy, "src/tools/clippy", "clippy-driver", stable=true, in_tree=true, {};
-    Miri, miri, "src/tools/miri", "miri", stable=false, {};
-    CargoMiri, miri, "src/tools/miri/cargo-miri", "cargo-miri", stable=false, {};
-    Rls, rls, "src/tools/rls", "rls", stable=true, {
+    Miri, miri, "src/tools/miri", "miri", stable=false, submodule="miri", {};
+    CargoMiri, miri, "src/tools/miri/cargo-miri", "cargo-miri", stable=false, submodule="miri", {};
+    Rls, rls, "src/tools/rls", "rls", stable=true, submodule="rls", {
         builder.ensure(Clippy {
             compiler: self.compiler,
             target: self.target,
             extra_features: Vec::new(),
         });
         self.extra_features.push("clippy".to_owned());
     };
-    Rustfmt, rustfmt, "src/tools/rustfmt", "rustfmt", stable=true, {};
-    RustAnalyzer, rust_analyzer, "src/tools/rust-analyzer/crates/rust-analyzer", "rust-analyzer", stable=false, {};
+    Rustfmt, rustfmt, "src/tools/rustfmt", "rustfmt", stable=true, submodule="rustfmt", {};
+    RustAnalyzer, rust_analyzer, "src/tools/rust-analyzer/crates/rust-analyzer", "rust-analyzer", stable=false, submodule="rust-analyzer", {};
```

Comment:
For now, these are redundant because they're also updated in bootstrap.py; I added them before I realized Cargo depended on the whole workspace being checked out. Let me know if I should drop this change.

Reply:
Yeah, it's probably best to avoid redundant updates to the submodules necessary for Cargo to run.

Reply:
Sure thing, I changed this to only manage `rust-analyzer`.